using_restricted_array.rb

```diff
@@ -38,29 +72,78 @@ def sort(array, length)
 # constant, adds an element with 'SPECIAL_VALUE' in the end. Assumes the array
 # to be sorted in ascending order.
 def delete(array, length, value_to_delete)
-  puts "NOT IMPLEMENTED"
+  i = 0
+
+  until i == length do
+    if array[i] == value_to_delete
+      array[i] = SPECIAL_VALUE

```

Comment:
This array is sorted in ascending manner and by convention we keep all SPECIAL_VALUEs in the end. Currently, the delete method is fragmenting your array. (e.g. if the array was [2, 6, 12, SPECIAL_VALUE] and value_to_delete was 6, the array will become [2, SPECIAL_VALUE, 12, SPECIAL_VALUE]). 
How can you update it so that all the SPECIAL_VALUEs remain in the end? And all used, valid values are in the beginning? (e.g. [2, 6, SPECIAL_VALUE, SPECIAL_VALUE])